### DISCUSSION TITLE
Autotagger

### DISCUSSION THREAD

**Discussion by captainpensive** [AprilSylph/XKit-Rewritten#225] [Suggestion Box]
Something which adds tags automatically, like an automatic queue tag, automatic video post tag, etc.

**Comment by AprilSylph** [comment]
MEGA EDIT: Auto Tagger's options are being folded into other features.

- **Keeping tags when reblogging**: Reimagined into "suggest tags from the post being reblogged" in Quick Reblog (v0.10.0)
- **Tagging who you're reblogging from**: Implemented into Quick Reblog's tag suggestions (v0.10.0)
- **Tagging the original post author**: Implemented into Quick Reblog's tag suggestions (v0.10.0)
- **Post types**: Implemented into Quick Reblog's tag suggestions (v0.10.0)
- **Queue tags**: Implemented into a text option in Quick Reblog (v0.11.0)
- **Original post tags**: Implemented into a text option in Quick Tags (v0.11.0)
- **Answer tags**: Implemented into a text option in Quick Tags (v0.11.0)

**Reply by irysova** [reply]
an automatic tag for the queue is useful by making it clear that we're not necessarily online when a post from the queue is posted! It would be awesome if y'all'd keep this feature in this xkit. but if not thanks anyway

**Reply by AprilSylph** [reply]
Tumblr has its own online indicators now, which you can enable or disable in your [Privacy Settings](https://www.tumblr.com/settings/privacy).

**Reply by irysova** [reply]
yes i don't want to announce i'm online tho.

**Reply by marcustyphoon** [reply]
There have been some discussions about this! In short: it's not off the table for the extension in the future, but it's off April's personal table at this moment.

(I personally like queue tags a lot. I suggest contacting Tumblr support and suggesting they be added as a native feature! It's one of those things that I think wouldn't feel out of place as a site feature.

If it never happens, I'll probably get to trying my hand at auto tagger at some point.)

**Reply by portablevampire** [reply]
> Tumblr has its own online indicators now, which you can enable or disable in your [Privacy Settings](https://www.tumblr.com/settings/privacy).

Besides Tumblr's online function being kind of garbage, I simply don't like using that as an online indicator, because I don't like broadcasting that I'm online. Queue tags as an online indicator are preferable to me because rather than broadcasting when I'm online, it tells people I may not necessarily be online at the moment, but doesn't definitively say whether I am or not. Just because you personally can't think of a reason to use queue tags doesn't mean there are none. 

I am currently using both Xkit Rewritten and New Xkit at the same time, specifically because having to manually tag every queue post is a pain for me, and I like having New Xkit's auto tagging for queue posts. I'd like to migrate fully to Xkit Rewritten since the auto tagging is the only thing I'm still using New Xkit for, but I won't be until this feature is implemented into Xkit Rewritten.

**Reply by AprilSylph** [reply]
If you're not ever broadcasting that you are online, why do you need to broadcast that you might not be?

**Reply by portablevampire** [reply]
> If you're not ever broadcasting that you are online, why do you need to broadcast that you might not be?

If you never broadcast whether or not you're online, people will tend to assume that if you're posting, you're online, which is a reasonable assumption to make. But I put a lot of things in my queue, and many of those things post during hours when I am asleep, or busy, etc, and if someone messages me during that time because they saw me posting and thought I might be online, I don't want them to think I am ignoring them. Sure, it's kind of on them for making that assumption, and I don't have an obligation to respond to people immediately even if I was online, but that's how I personally prefer to operate.

Essentially: broadcasting that I'm online makes me feel obligated to respond to people, even if I'm not. But people will assume I'm online if they see me posting regardless, so I want a way to give plausible deniability to my online status.

Furthermore, I just like having my queue posts marked separately from my non-queued posts. Perhaps that's in some part a result of my OCD, but it's a form of organization that is comforting to me, and I feel like the option should be there.

**Comment by lefae** [comment]
From both a chronically ill (and thus in need of anything that helps make tumblr easier to use for me) as well as a roleplayer perspective, the best part of autotagger was the ability to automatically tag the person you were reblogging from, as well as automatically replacing any hyphens in their url with spaces, as this makes it easier for people to not only find their own threads but to be able to blacklist anything they don't want to see. Having the extension handle that for me certainly made things simplier where that much is concerned, since some days even tagging posts when I draft them takes more energy than I might currently have available.

On that same point: a queue tag can be beneficial depending on how the blogger uses theirs. I personally use mine specifically for "filler posts" (ie aesthetics, gifsets, fandom related stuff, etc) so that if someone wants to blacklist the queue tag, they can and not miss out on the active content where my own writing and such is concerned. But that's very much a ymmv situation, since everyone seems to use their queue differently.

**Reply by jillyfae** [reply]
A significant proportion of the people I follow use(d) the auto-queue tag of some sort for similar reasons too, as well as the post type for other people to use to filter their dashboard. (Especially video posts, since some of the other "stop autoplay" features don't always work, so filtering all video posts until you click on them can help a lot.)

Autotagger was one of my most consistently used features on the previous xkit. Queue tag to differentiate /stuff I randomly liked and am sharing/ versus stuff I was *talking about*, source tags for art/fic, video & audio post tags, etc.

**Reply by AprilSylph** [reply]
I'm implementing semi-automatic tagging of the blog you're reblogging from (as well as the original post author) as part of Quick Reblog's new tag suggestions feature landing in v0.10.0. Hyphens have no need to be replaced since they can now be parsed correctly by Tumblr's [better tagged and search URL decoding](https://staff.tumblr.com/post/668748027469676544/hello-for-years-multi-word-tags-and-tags-that).

**Comment by slach123** [comment]
I recognize that different post types no longer exist so the large bulk of what autotagger did isn't necessary, but I did find original post tagging and queued post tagging very valuable. If they get folded into another feature, that's fine, but I really hope that queued post tags continue to be an option, even if you yourself don't think they're valuable, a lot of people do like them. I personally use the queue tag because I "time travel" a lot on my dash and queue those older posts so then when I'm trying to catch up on my dash, I can distinguish between when I was actually on my dash vs when I was at a different time based on my reblogs. I know there are dim old posts and the "you're caught up" carousel, but those don't work very well when you time travel a lot and have multiple tabs of tumblr up.

Even if it's not something you use or think is necessarily handy, it would be a nice feature for others, and I hope you'll consider it for the future. I will also be talking to tumblr support and seeing if they will implement it natively, but I don't have high hopes for that.

**Reply by AprilSylph** [reply]
Original post tagging will be present as a field in Quick Tags in v0.11.0. A queue tag field will also be present in Quick Reblog in the same release.

**Comment by kelparoo** [comment]
Is it possible to get these two features (Keeping tags when reblogging, Tagging who you're reblogging) from the quick reblogs quick tagging to also work in the normal reblogging? As an RPer I often open the post immediately as a normal reblog so I can write replies, and it's much easier when the tags are already there, especially when they have long or hard to spell urls, or say something i need to remember to reply to in tags. with the current method, i need to hove over quick reblog, use the suggested tags to save any tags i want to keep, then quick draft it before navigating to drafts and replying from there. whereas with the old xkit auto tagger (which is not working anymore for some reason) it worked even in normal reblogs and was one of the main functions for rp blogs.
